src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1405,6 +1438,203 @@ private void WritePackageDependencies()
                 }
             }
 
+            // This implementation was taken from PreprocessPackageDependenciesDesignTime.ExecuteCore
+            // with some minor modifications to use WriteItem instead of saving it into an array.
+            private void WritePackageDependenciesDesignTime()
```

Comment:
I think we can simplify this method (and the nested methods within it) more.  In `GetPackageDefinitions` we are looping through the lock file libraries and creating a `PackageDefinitionItem` for each one.  Then in the top-level method we are looping through those PackageDefinitionItems, skipping the ones that aren't top-level dependencies, and transforming them into the items that we write out.

Can we try getting rid of the `GetPackageDefinitions` method and the intermediate `PackageDefinitionItem` type, and combine this logic into one loop in the top-level method?

Reply:
done